### PR TITLE
MaterialOutputAction Refactor

### PR DESCRIPTION
**Pull request by permcody**
This PR refactors MaterialOutputAction so that it no longer adds objects during the wrong tasks. Additionally it is no longer a meta-action.

### CODE REVIEW

File: framework/src/actions/MaterialOutputAction.C [hunk 1]
```diff
@@ -57,28 +59,22 @@ MaterialOutputAction::MaterialOutputAction(InputParameters params)
...
-
-  buildMaterialOutputObjects(_problem.get());
-}
+  mooseAssert(_problem,
+              "FEProblemBase pointer is nullptr, it is needed for auto material property output");
 
-void
-MaterialOutputAction::buildMaterialOutputObjects(FEProblemBase * problem_ptr)
-{
   // Do nothing if the application does not have output
```
Comment (aeslaughter) [inline framework/src/actions/MaterialOutputAction.C]: This should get removed, it is possible to have Output objects without "add_output" action.

File: framework/src/actions/MaterialOutputAction.C [hunk 2]
```diff
@@ -93,7 +89,8 @@ MaterialOutputAction::buildMaterialOutputObjects(FEProblemBase * problem_ptr)
   {
```
Comment (aeslaughter) [inline framework/src/actions/MaterialOutputAction.C]: As far as I can tell from the problem @sapitts is having, this loop is the actual problem. We need to loop over actual Output objects. She is adding Output objects via FEProblemBase::addOutput, so they don't show up in this action-based list, thus directing VPP objects to the output fails.

File: framework/src/actions/MaterialOutputAction.C (hunk 1 above)
Comment (permcody) [reply framework/src/actions/MaterialOutputAction.C]: right - removed!

File: framework/src/actions/MaterialOutputAction.C (hunk 2 above)
Comment (permcody) [reply framework/src/actions/MaterialOutputAction.C]: As we discussed any Action may satisfy the add_outputs task, and if they register with that task, they will be returned with the rest of the built-in Actions. This should be working.

**Comment by permcody** [comment]
@aeslaughter - This looks like it's finally ready for a review.

@sapitts - When this gets merged, I can assist you in getting your Action working.

**Review by aeslaughter — APPROVED** [review]

**Pull request merged**